src/styles/blocks/main.scss

```diff
@@ -0,0 +1,8 @@
+.main {
+  padding: 50px 40px;
+
+  display: grid;
+  grid-template-columns: repeat(auto-fill, 200px);
```

Comment:
The `grid-template-columns` property should be adjusted to set the number of columns based on screen width. Consider using media queries to define different column settings for various screen sizes as per the task requirements.